ppl/cmd/zqd/listen/command.go

```diff
@@ -113,15 +114,26 @@ func (c *Command) Run(args []string) error {
 	if err != nil {
 		return err
 	}
-	core.Registry().MustRegister(prometheus.NewGoCollector())
 	c.logger.Info("Starting",
 		zap.String("datadir", c.conf.Root),
 		zap.Uint64("open_files_limit", openFilesLimit),
+		zap.String("personality", c.personality),
 		zap.Bool("pprof_routes", c.pprof),
 		zap.Bool("suricata_supported", core.HasSuricata()),
 		zap.Bool("zeek_supported", core.HasZeek()),
 	)
-	h := zqd.NewHandler(core, c.logger)
+	switch c.personality {
+	case "all":
+		core.AddAPIServerRoutes()
+		core.AddWorkerRoutes()
+	case "apiserver":
+		core.AddAPIServerRoutes()
+	case "worker":
+		core.AddWorkerRoutes()
+	default:
+		return fmt.Errorf("unknown personality %s", c.personality)
+	}
```

Comment:
I think it would be better to not call these functions in order for core to server any routes. Ideally it would by default have personality "all". Perhaps make personality a variable in config and move this switch to NewCore?

Reply:
@mattnibs: Like 863a2edab8eeab8743fe818d401ddf76eec15aa9?

Reply:
Excellent